src/unix/os390-syscalls.c

```diff
@@ -0,0 +1,372 @@
+    struct pollfd pfd;
+
+    pfd.fd = fd;
+    uv_mutex_lock(&lst->lock);
+    found = lfind(&pfd, &lst->items[0], &lst->size,
+                  sizeof(struct pollfd), isfdequal);
+    if (found != NULL)
+      memcpy(found,
+             found+1,
+             (&lst->items[lst->size--] - found - 1) * sizeof(*found));
+    uv_mutex_unlock(&lst->lock);
+
+    if (found == NULL)
+      return ENOENT;
+
+  }
+  else if(op == EPOLL_CTL_ADD)
+  {
+    struct pollfd* found;
+    struct pollfd pfd;
+    size_t before;
+    size_t after;
+
+    if (lst->size == MAX_ITEMS_PER_EPOLL - 1)
+      return ENOMEM;
+
+    pfd.fd = fd;
+    pfd.events = event->events;
+    uv_mutex_lock(&lst->lock);
+    before = lst->size; 
+    found = lsearch(&pfd, &lst->items[0], &lst->size,
+                    sizeof(struct pollfd), isfdequal);
+    after = lst->size; 
+    uv_mutex_unlock(&lst->lock);
+
+    if (found != NULL && before == after) {
+      errno = EEXIST;
+      return -1;
+    }
+
+  }
+  else if(op == EPOLL_CTL_MOD)
+  {
+    int index;
+    struct pollfd* found;
+    struct pollfd pfd;
+
+    pfd.fd = fd;
+    uv_mutex_lock(&lst->lock);
+    found = lfind(&pfd, &lst->items[0], &lst->size,
+                  sizeof(struct pollfd), isfdequal);
+
+    if (found != NULL)
+      found->events = event->events;
+      
+    uv_mutex_unlock(&lst->lock);
+
+    if (found == NULL) {
+      errno = ENOENT;
+      return -1;
+    }
+  }
+  else 
+  {
+    abort();
+  }
+  return 0;
+}
+
+int epoll_wait(int epfd, struct epoll_event *events, int maxevents, int timeout)
+{
+  struct _epoll_list *lst = _global_epoll_list[epfd];
+
+  unsigned int size = lst->size;
+
+  struct pollfd *pfds = lst->items;
+  int returnval = poll( pfds, size, timeout );
+  if(returnval == -1)
+    return returnval;
+
+  int reventcount=0;
+  int realsize = lst->size;
+  for (int i = 0; i < realsize && i < maxevents; ++i)                     
+  {
+    struct epoll_event ev = { 0, 0 };
+    ev.fd = pfds[i].fd;
+    if(!pfds[i].revents)
+      continue;
+
+    if(pfds[i].revents & POLLRDNORM)
+      ev.events = ev.events | POLLIN;
+
+    if(pfds[i].revents & POLLWRNORM)
+      ev.events = ev.events | POLLOUT;
+
+    if(pfds[i].revents & POLLHUP)
+      ev.events = ev.events | POLLHUP;
+
+    pfds[i].revents = 0;
+    events[reventcount++] = ev; 
+
+  }
+
+  return reventcount;
+}
+
+int epoll_file_close(int fd)
+{
+  for( int i = 0; i < number_of_epolls; ++i )
+  {
+    struct _epoll_list *lst = _global_epoll_list[i];
+    struct pollfd* found;
+    struct pollfd pfd;
+
+    pfd.fd = fd;
+    uv_mutex_lock(&lst->lock);
+    found = lfind(&pfd, &lst->items[0], &lst->size,
+                  sizeof(struct pollfd), isfdequal);
+    if (found != NULL)
+      memcpy(found,
+             found+1,
+             (&lst->items[lst->size--] - found - 1) * sizeof(*found));
+    uv_mutex_unlock(&lst->lock);
+  }
+
+  return 0;
+}
+
+int getexe(const int pid, char *buf, size_t len) {
```

Comment:
I did that for code readability.
I wanted to separate the uv_exepath which uses 3 high level  strategies to obtain absolute path using posix apis from getexe which retrieves the stored relative path using non-posix zos syscall.
I will bring getexe into the same file and declare as static